collections-csharp/UpdateItemInListCSharp/src/UpdateItemInListCSharp/Models/Library.cs

```diff
@@ -0,0 +1,81 @@
+﻿namespace UpdateItemInListCSharp.Models
+{
+    public class Library
+    {
+        public List<Book> Books { get; set; }
+
+        public Library()
+        {
+            Books =
+                [
+                new("J.K. Rowling", "978-0439708180", false, "Harry Potter and the Philosopher's Stone"),
+                new("J.K. Rowling", "978-0439064873", false, "Harry Potter and the Chamber of Secrets"),
+                new("J.K. Rowling", "978-0439139601", false, "Harry Potter and the Goblet of Fire"),
+                new("J.K. Rowling", "978-0439358071", false, "Harry Potter and the Order of the Phoenix"),
+                new("J.K. Rowling", "978-0439785969", false, "Harry Potter and the Half-Blood Prince")
+            ];
+        }
+
+        public Book AddBook(string author, string isbn, bool isCheckedOut, string title)
+        {
+            var book = new Book(author, isbn, isCheckedOut, title); Books.Add(book);
+            return book;
+        }
+
+        public void CheckoutBookFind(string isbn)
+        {
+            var bookToCheckout = Books.Find(b => b.ISBN == isbn);
+            if (bookToCheckout != null)
+            {
+                bookToCheckout.IsCheckedOut = true;
+            }
+        }
+
+        public void CheckoutBookFindIndex(string isbn)
+        {
+            var index = Books.FindIndex(b => b.ISBN == isbn);
+            if (index != -1)
+            {
+                Books[index].IsCheckedOut = true;
+            }
+        }
+
+        public void CheckoutBookFirstOrDefault(string isbn)
+        {
+            var bookToCheckoutFirstOrDefault = Books.FirstOrDefault(b => b.ISBN == isbn);
+            if (bookToCheckoutFirstOrDefault != null)
+            {
+                bookToCheckoutFirstOrDefault.IsCheckedOut = true;
+            }
+        }
+
+        public void CheckoutBookForeach(string isbn)
+        {
+            foreach (var bookItem in Books)
+            {
+                if (bookItem.ISBN == isbn)
+                {
+                    bookItem.IsCheckedOut = true;
+                }
+            }
+        }
+
+        public void CheckoutBookIndexOf(Book book)
```

Comment:
This is not the correct way to get the index of the matching book object. Object is by reference, since you are adding the same object in the list and pass the same book object in the list, it is working for you.

Reply:
I don't quite understand this point. I add a book and then check out the same book using the CheckoutBookUsingIndexOf method. In my test method for this method, I get a new object from this book, which is also checked out. So I have achieved exactly what the method was supposed to do.

Reply:
I update the book class and the CheckoutUsingIndexOf() method.
These changes allow me to find a book in the list based on its properties rather than the specific object instance (reference).